tests/main/install-hook/task.yaml

```diff
@@ -0,0 +1,48 @@
+summary: Check install and remove hooks.
+
+execute: |
+    . $TESTSLIB/snaps.sh
+    install_local snap-hooks
+
+    echo "Verify configuration value with snap get"
+    snap get snap-hooks installed | MATCH 1
+    snap get snap-hooks foo | MATCH bar
+
+    echo "Verify that install hook is run only once"
+    snap set snap-hooks installed=2
+    install_local snap-hooks
+    snap get snap-hooks installed | MATCH 2
+
+    REMOVE_HOOK_FILE=/root/snap/snap-hooks/current/remove-hook-executed
+
+    echo "Verify that remove hook is not executed when removing single revision"
+    snap remove --revision=x1 snap-hooks
+    if test -f $REMOVE_HOOK_FILE; then
+        echo "Remove hook was executed. It shouldn't."
+        exit 1
+    fi
+
+    echo "Verify that remove hook is executed"
+    snap remove snap-hooks
+    if ! test -f $REMOVE_HOOK_FILE; then
+        echo "Remove hook was not executed"
+        exit 1
+    fi
+
+    echo "Installing a snap with hooks again"
+    rm -rf "$REMOVE_HOOK_FILE" 2>&1 > /dev/null
+    install_local snap-hooks
+
+    echo "Forcing remove script to fail"
+    snap set snap-hooks exitcode=1
+    snap remove snap-hooks
```

Comment:
No, it wouldn't fail on non-zero because of the setup of this hook has IgnoreError set. However, it's always 0 because of the problem with configure you spotted below - thanks! I've improved this test to make sure remove hook sees the correct exitcode value.